oauth2-authorization-server/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationCodeGrantTests.java

```diff
@@ -294,18 +299,14 @@ public void requestWhenPublicClientWithPkceThenReturnAccessTokenResponse() throw
 				.params(getTokenRequestParameters(registeredClient, authorization))
 				.param(OAuth2ParameterNames.CLIENT_ID, registeredClient.getClientId())
 				.param(PkceParameterNames.CODE_VERIFIER, S256_CODE_VERIFIER))
+				.andExpect(header().string(HttpHeaders.CACHE_CONTROL, containsString("no-store")))
+				.andExpect(header().string(HttpHeaders.PRAGMA, containsString("no-cache")))
 				.andExpect(status().isOk())
 				.andExpect(jsonPath("$.access_token").isNotEmpty())
 				.andExpect(jsonPath("$.token_type").isNotEmpty())
 				.andExpect(jsonPath("$.expires_in").isNotEmpty())
 				.andExpect(jsonPath("$.refresh_token").doesNotExist())
 				.andExpect(jsonPath("$.scope").isNotEmpty());
-
-		verify(registeredClientRepository, times(2)).findByClientId(eq(registeredClient.getClientId()));
```

Comment:
Is there a reason these `verify()`'s were removed?

Reply:
We briefly touched on this earlier, the idea of this test (`requestWhenPublicClientWithPkceThenReturnAccessTokenResponse`) is to verify that the end-to-end PKCE happy path works as expected.

As such, there are two requests made and therefore we call the `registeredClientRepository.findByClientId` twice. But:
- This implementation detail is irrelevant to PKCE itself
- `times(2)` might be confusing to someone reading the tests
- The fact that `registeredClientRepository.findByClientId` is only called once in `/authorize` and only once in `/token` is already covered by other tests, e.g.
  - `requestWhenAuthorizationRequestAuthenticatedThenRedirectToClient` for `/authorize`
  - `requestWhenTokenRequestValidThenReturnAccessTokenResponse` for `/token`


Maybe this should be done in another PR?